File: LightningConstructs/Source/Socket/Declarations.cs
namespace Lightning
{
    public partial class Socket
    {
        private System.Net.Sockets.Socket NetSocket = new System.Net.Sockets.Socket(System.Net.Sockets.SocketType.Stream, System.Net.Sockets.ProtocolType.Tcp);
        private System.Boolean IsListening = false;
        private System.Boolean IsConnected = false;
        private System.Boolean IsBaseDisposed = false;
        private Lightning.Governor ListenerGovernor = new Lightning.Governor(20);

        private System.Object ReadLock = new System.Object();
        private System.Object WriteLock = new System.Object();
    }
}

File: LightningConstructs/Source/Socket/Declarations.vb
    Public Partial Class Socket
        Private NetSocket As Net.Sockets.Socket = New Net.Sockets.Socket(Net.Sockets.SocketType.Stream, Net.Sockets.ProtocolType.Tcp)
        Private IsListening As System.Boolean = False
        Private IsConnected As System.Boolean = False
        Private IsBaseDisposed As System.Boolean = False
        Private ReadOnly ListenerGovernor As Governor = New Governor(10)
        Private DataAvailable As Boolean = False
        Private DataNotAvailable As Boolean = False
        Private BaseClientBound As Boolean = False
        
        Private ReadOnly ReadLock As System.Object = New System.Object()
        Private ReadTimeoutStopwatch As Stopwatch = New Stopwatch()
        
        Private ReadOnly WriteLock As System.Object = New System.Object()
    End Class
